File: stack.py


class Stack:
  """ A simple stack class """
  def __init__(self):
    self.arr = []

  def push(self, elem):
    self.arr.insert(-1, elem)

  def pop(self):
    return self.arr.pop()

  def top(self):
    return self.arr[-1]

File: stack.rb

class Stack

  def initialize(size=0, type=Object)
    @arr = Array.new(size, type)
  end

  def push(elem)
    @arr.push(elem)
  end

  def pop
    @arr.pop
  end

  def top
    @arr.last
  end
  
end
